src/profiling/meter_manager.cpp

```diff
@@ -0,0 +1,71 @@
+#include "meter_manager.hpp"
+
+namespace nest {
+namespace mc {
+namespace util {
+
+using nlohmann::json;
+
+meter_manager::meter_manager() {
+    // add time-measurement meter
+    meters_.emplace_back(new time_meter());
+
+    // add memory consumption meter
+    // TODO
+
+    // add energy consumption meter
+    // TODO
+};
+
+void meter_manager::checkpoint(std::string name) {
+    checkpoint_names_.push_back(std::move(name));
+
+    // Enforce a global synchronization point the first time that the meters
+    // are used, to ensure that times measured across all domains are
+    // synchronised.
+    if (meters_.size()==0) {
```

Comment:
erm, good catch. That should be `if (checkpoint_names_.size()==1)`, because we only want to do the synchronization if this is the first checkpoint.